0000-product.md

```diff
@@ -0,0 +1,463 @@
+
+The Grid Product implementation is designed for sharing product master data
+between participants. Product is a near universal concept within supply chain
+solutions and would naturally be one of the highest areas of re-use across Grid
+applications.
+
+The desgn will address use cases including:
+
+- Sharing of Product master data across a network
+- Including Product in other business transactions (track and trace events,
+  purchases, sales, etc)
+- Enriching UX experiences by including additional attribution of a Product such
+  as names or descriptions
+
+It is also useful to use product information as auxiliary data in other supply
+chain solutions.  For example, in track and trace, the location and temperature
+of an item (an instance of a product) is stored.  When presenting this
+information later to a human user, it is useful to combine it with information
+about the product, such as the product’s name and description.  Therefore,
+Product becomes a reusable and important component within Grid.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A **__product__** is an archetype of an item that is transacted, traded, or
+referenced in a supply chain.  Each product has a **__product_namespace__**.
+This RFC defines a single product_namespace for GS1; a product in the GS1
+product_namespace is called a GS1 product.  Note that this design supports
+extending additional product namespaces in the future.
+
+A product is referenced using a **product_id**.  For GS1 products, the
+product_id is a Global Trade Item Number (GTIN), which is part of the GS1
+specifications.
+
+Each product has an **__owner__**, which is the identifier of the organization
+responsible for maintaining the product.  An **__agent__** acts on behalf of an
+organization, and all permissions are defined in terms of the actions agents can
+perform.  Organizations and agents are defined and managed by Pike (another
+component of Grid).
+
+A product has one or more **properties**.  Properties are described in the Grid
+Primitives RFC.  A *property namespace* contains multiple *property schemas*.  A
+property schema associates a name (such as “length”) with a data type (such as
+integer).  GS1 products may only include properties defined in the GS1 product
+property namespace.
+
+## Transactions
+
+Products are managed by submitting transactions to Hyperledger Grid, which will
+process them with the Grid Product smart contract. The following transactions
+are supported:
+
+* ProductCreate - create a Product and store it in state.  
+* ProductUpdate - update (replace) the properties of a Product already in
+  state.
+* ProductDelete - remove a Product from state.
+
+## Permissions
+
+Creation of GS1 products is restricted to agents acting on behalf of the
+organization that is associated with the GS1 company prefix encoded in the GTIN.
+An organization must have the GS1 company prefix in the “gs1_company_prefixes”
```

Comment:
Should
 this be company prefix or an Entity GLN (which has associated master data?)

Reply:
The purpose of this check is to ensure the organization creating the product, has the permission to do so. The GS1 company prefix is in the GTIN, so to ensure that the org (company) has the permission it must have an additional field in its metadata titled "gs1_company_prefix" with that same value in the GTIN.

Is that the correct method for validation of permissions?

Reply:
I understand, and I think so. It is just that organizations identity themselves to one another not by company prefix but by GLN (which has the same co prefix as the product). Need to think about it some more - leave as is for now